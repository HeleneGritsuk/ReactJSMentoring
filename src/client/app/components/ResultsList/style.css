.searchResults__content {
  display: flex;
  padding: 40px 0;
  position: relative;
  flex-wrap: wrap;
}

.searchResults__message {
  position: absolute;
  top: 50%;
  left: 50%;
  font-weight: bold;
  font-size: 20px;
}

.filmItem {
  margin: 0 1.1% 35px;
  width: 31%;
  text-decoration: none;
  color: #000;
}
.filmItem:hover .filmItem__img {
  box-shadow: 7px 8px 11px 1px rgba(0,0,0,0.16);
}
.filmItem__info {
  display: flex;
  margin-top: 10px;
}

.filmItem__year {
  margin-left: auto;
}
.filmItem__year span{
  padding: 5px;
  border: 1px solid #ccc;
  border-radius: 5px;
  line-height: 23px;
}
.filmItem__img {
  width: 100%;
  height: auto;
  border-radius: 6px;
}
.filmItem__title {
  font-weight: bold;
  text-transform: uppercase;
  font-size: 17px;
}
.filmItem__genre {
  color: #666;
}
