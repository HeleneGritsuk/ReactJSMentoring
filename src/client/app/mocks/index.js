const filmsMockArray = [
  {
    title: "Kill Bill",
    poster_path:
      "http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg",
    release_date: 2014,
    id: 1,
    genres: ["Action", "Adventure", "Comedy"]
  },
  {
    title: "The Wolf of Wall Street",
    poster_path:
      "https://image.tmdb.org/t/p/w500/vK1o5rZGqxyovfIhZyMELhk03wO.jpg",
    release_date: "2013-12-25",
    id: 2,
    genres: ["Crime", "Drama", "Comedy"]
  }
];
const filmMock = {
  budget: 104000000,
  genres: ["Action", "Adventure", "Comedy"],
  id: 343668,
  overview:
    "When an attack on the Kingsman headquarters takes place and a new villain rises, Eggsy and Merlin are forced to work together with the American agency known as the Statesman to save the world.",
  poster_path:
    "https://image.tmdb.org/t/p/w500/34xBL6BXNYFqtHO9zhcgoakS4aP.jpg",
  release_date: "2017-09-20",
  revenue: 394247609,
  runtime: 141,
  tagline: "Reports of my death have been greatly exaggerated.",
  title: "Kingsman: The Golden Circle",
  vote_average: 7,
  vote_count: 319
};
const filmInfoMock = {
  budget: 104000000,
  genres: (3)[("Action", "Adventure", "Comedy")],
  id: 343668,
  overview:
    "When an attack on the Kingsman headquarters takes place and a new villain rises, Eggsy and Merlin are forced to work together with the American agency known as the Statesman to save the world.",
  poster_path:
    "https://image.tmdb.org/t/p/w500/34xBL6BXNYFqtHO9zhcgoakS4aP.jpg",
  release_date: "2017-09-20",
  revenue: 394247609,
  relatedFilms: [
    {
      budget: 104000000,
      genres: (3)[("Action", "Adventure", "Comedy")],
      id: 343668,
      overview:
        "When an attack on the Kingsman headquarters takes place and a new villain rises, Eggsy and Merlin are forced to work together with the American agency known as the Statesman to save the world.",
      poster_path:
        "https://image.tmdb.org/t/p/w500/34xBL6BXNYFqtHO9zhcgoakS4aP.jpg",
      release_date: "2017-09-20",
      revenue: 394247609,
      runtime: 141,
      tagline: "Reports of my death have been greatly exaggerated.",
      title: "Kingsman: The Golden Circle",
      vote_average: 7,
      vote_count: 319
    },
    {
      budget: 104000000,
      genres: (3)[("Action", "Adventure", "Comedy")],
      id: 343668,
      overview:
        "When an attack on the Kingsman headquarters takes place and a new villain rises, Eggsy and Merlin are forced to work together with the American agency known as the Statesman to save the world.",
      poster_path:
        "https://image.tmdb.org/t/p/w500/34xBL6BXNYFqtHO9zhcgoakS4aP.jpg",
      release_date: "2017-09-20",
      revenue: 394247609,
      runtime: 141,
      tagline: "Reports of my death have been greatly exaggerated.",
      title: "Kingsman: The Golden Circle",
      vote_average: 7,
      vote_count: 319
    }
  ],
  runtime: 141,
  tagline: "Reports of my death have been greatly exaggerated.",
  title: "Kingsman: The Golden Circle",
  vote_average: 7,
  vote_count: 319
};

export { filmsMockArray, filmMock, filmInfoMock };
