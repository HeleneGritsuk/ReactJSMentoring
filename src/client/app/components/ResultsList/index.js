import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

const ResultsList = ({ films, getFilmInfo }) => {
  if (films.length) {
    return (
      <div className="searchResults__content container">
        {
            films.map(film =>
            (
              <Link to={`/film/${film.id}`} className="filmItem" key={film.id} >
                <img className="filmItem__img" src={film.poster_path} />
                <div className="filmItem__info">
                  <div className="filmItem__title">
                    {film.title}
                  </div>
                  <div className="filmItem__year">
                    <span>{new Date(film.release_date).getFullYear()}</span>
                  </div>
                </div>
                <div className="filmItem__genre">
                  {film.genres.join(' & ')}
                </div>
              </Link>))}
      </div>
    );
  }
  return (
    <div className="searchResults__content container">
      <span className="searchResults__message">No films found</span>
    </div>
  );
};

export default ResultsList;
