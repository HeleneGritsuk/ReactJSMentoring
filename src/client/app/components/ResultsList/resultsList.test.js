import React from 'react';
import ResultsList from './index';
import renderer from 'react-test-renderer';
import { filmsMockArray } from '../../mocks';
import { text, shallow } from 'enzyme';

test('ResultList should render list of films', () => {
  const component = renderer.create(
    <ResultsList films = { filmsMockArray }/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('ResultList shoud display text "No films found" when there is no films found', () => {
  const component = shallow(
    <ResultsList films = { [] }/>,
  );
  expect(component.text()).toEqual('No films found')
});
