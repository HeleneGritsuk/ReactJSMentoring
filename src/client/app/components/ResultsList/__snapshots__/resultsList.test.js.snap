// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList should render list of films 1`] = `
<div
  className="searchResults__content"
>
  <div
    className="filmItem"
  >
    <img
      src="http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg"
    />
    <div
      className="filmItem__info"
    >
      <div
        className="filmItem__title"
      >
        Kill Bill
      </div>
      <div
        className="filmItem__year"
      >
        2014
      </div>
    </div>
    <div
      className="filmItem__genre"
    >
      drama
    </div>
  </div>
  <div
    className="filmItem"
  >
    <img
      src="http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg"
    />
    <div
      className="filmItem__info"
    >
      <div
        className="filmItem__title"
      >
        Kill Bill2
      </div>
      <div
        className="filmItem__year"
      >
        2017
      </div>
    </div>
    <div
      className="filmItem__genre"
    >
      adventure
    </div>
  </div>
</div>
`;
